hedyorg/hedy
# [FIX] Fixes skip faulty warning being shown

**Description**

When the server takes too long to respond to a  `/parse` request, the front-end assumes that this must be because it is recompiling the program after finding errors, due to this code in `app.ts`

```
let errorsFoundTimeout = setTimeout(function () {
        error.showWarningSpinner();
        error.showWarning(ClientMessages['Execute_error'], ClientMessages['Errors_found']);
      }, 500)

let response = await postJsonWithAchievements('/parse', data);
clearTimeout(errorsFoundTimeout);
error.hide()
```
This essentially means: wait for 500 ms for response, if it answers before you don't execute the function and hides any possible errors. I changed that and now we only show the warning box if we find errors we skipped in the source map. This means that we are not relying on the response time of the server, and will reliably show the warning. 

I changed also the wording of the warning and now it says this: "You made a mistake! Don't worry, we still ran the program". However, @Felienne can you tell me if this is a good wording, I'm not sure this is clear enough.

**How to test**

Easiest way to reproduce is:

1. add a sleep to `/parse` endpoint
2. See that now even if the server takes long to respond the skip faulty error is not shown

Depends-On: #4560
